actions/dependabot-auto-triage/src/index.ts

```diff
@@ -0,0 +1,260 @@
+        ) {
+          console.error(`
+ERROR: Cannot access Dependabot alerts.
+
+This error may occur for several reasons:
+1. Dependabot alerts might not be enabled for this repository
+   - Enable them at: https://github.com/${owner}/${repo}/settings/security_analysis
+   
+2. The token being used might not have sufficient permissions
+   - For GITHUB_TOKEN: Make sure the workflow has 'security-events: write' permission
+   - For fine-grained tokens: Make sure the 'Dependabot alerts: write' permission is enabled
+
+3. Organization policies might be restricting access to security features
+   - Check with your organization administrator
+
+If you're sure the token has the right permissions, try accessing the alerts manually at:
+https://github.com/${owner}/${repo}/security/dependabot
+          `);
+          process.exit(1);
+        } else {
+          console.error(`API Error ${error.status}: ${error.message}`);
+          process.exit(1);
+        }
+      } else {
+        console.error(
+          "Error:",
+          error instanceof Error ? error.message : String(error),
+        );
+        process.exit(1);
+      }
+    }
+  } catch (error) {
+    console.error(
+      "Error:",
+      error instanceof Error ? error.message : String(error),
+    );
+    process.exit(1);
+  }
+}
+
+async function fetchAllAlerts(
+  octokit: Octokit,
+  owner: string,
+  repo: string,
+  alertTypes: string[],
+) {
+  const alerts = [];
+  let page = 1;
+  let hasMorePages = true;
+
+  try {
+    while (hasMorePages) {
+      // Using the dependabot alerts endpoint directly
+      const response = await octokit.request(
+        "GET /repos/{owner}/{repo}/dependabot/alerts",
+        {
+          owner,
+          repo,
+          state: "open",
+          per_page: 100,
+          page,
+        },
+      );
+
+      if (response.data.length === 0) {
+        hasMorePages = false;
+      } else {
+        const filteredAlerts = response.data.filter(
+          (alert) =>
+            alertTypes.includes(alert.security_advisory?.severity as string) ||
+            alertTypes.includes(
+              alert.security_vulnerability?.severity as string,
+            ) ||
+            alertTypes.includes("dependency"),
+        );
+        alerts.push(...filteredAlerts);
+        page++;
+      }
+    }
+
+    return alerts;
+  } catch (error) {
+    // Re-throw to be handled by the main function
+    throw error;
+  }
+}
+
+function matchesAnyPattern(
+  manifestPath: string | undefined,
+  patterns: string[],
+): boolean {
+  if (!manifestPath) return false;
+
+  return patterns.some((pattern) => {
+    try {
+      return minimatch(manifestPath, pattern, { matchBase: true });
+    } catch (error) {
+      console.error(
+        `Error matching pattern ${pattern}:`,
+        error instanceof Error ? error.message : String(error),
+      );
+      return false;
+    }
+  });
+}
+
+run();
```

Comment:
```suggestion
if (import.meta.main) {
  run();
}
```

So that you can also run unittests against that file without executing the `run` function by accident 🙂

Reply:
Done!